src/internal/clusterstate/v2.11.0/pjs.go

```diff
@@ -0,0 +1,55 @@
+package v2_11_0
+
+import (
+	"context"
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/migrations"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pctx"
+)
+
+func createPJSSchema(ctx context.Context, env migrations.Env) error {
+	ctx = pctx.Child(ctx, "createPJSSchema")
+	tx := env.Tx
+	_, err := tx.ExecContext(ctx, `
+		CREATE SCHEMA pjs; 
+		CREATE TYPE pjs.job_error_code AS ENUM (
+			'failed', 
+			'disconnected',
+			'canceled'
+		);
+		CREATE TABLE pjs.jobs (
+				id BIGSERIAL PRIMARY KEY,
+				parent BIGINT REFERENCES pjs.jobs(id),
+				
+				program BYTEA NOT NULL,
+				program_hash BYTEA NOT NULL, -- the hash will be used to model the queues
+				
+				error pjs.job_error_code,
+				
+				queued timestamptz DEFAULT CURRENT_TIMESTAMP NOT NULL,
+				processing timestamptz,
+				done timestamptz		
+		);
+		CREATE TYPE pjs.fileset_types AS ENUM (
```

Comment:
An alternative is 2 separate tables.  I don't know how that affects performance.  I would do whatever is easier to write queries against for now.